### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by mnordine** [bp74/StageXL#257]
This will leak until it crashes the browser:

```dart
import 'dart:html';
import 'dart:async' show Future;
import 'package:stagexl/stagexl.dart';

main() async
{
  final canvas = new CanvasElement(width: 800, height: 600);

  document.body.children.add(canvas);

  final stage = new Stage(canvas);

  new RenderLoop()..addStage(stage);

  while (true) (await loadResources()).dispose();
}

Future<ResourceManager> loadResources() async
{
  final resources = new ResourceManager();

  for (var i = 0; i < 5; i++)
    resources.addTextureAtlas('assets_$i', 'texture_atlases/assets_$i@1x.json');

  await resources.load();

  return resources;
}
```

**Comment by mnordine** [comment]
Note that GPU memory is reclaimed perfectly.

**Comment by mnordine** [comment]
In Chrome Dev Tools, Used JS Heap, Nodes, Listeners all continue to climb.

![mem](https://cloud.githubusercontent.com/assets/1924573/20637356/debdd942-b35b-11e6-9d8b-dc6fd3b53a76.PNG)

**Comment by mnordine** [comment]
My initial hunch was that an `ImageElement` was created in each `ImageLoader`, and that's hanging around. I tried adding a `dispose` to `ImageLoader`, but that didn't work:

```dart
void dispose()
{
  image?.remove();
  image?.src = '';
}
```

**Comment by bp74** [comment]
This is very interesting. It looks like as if the event listeners on the image element are not garbage collected. The number of Listeners is increasing over time. But both event listeners on the image element are canceled once the image is loaded. Will do some further investigations ...

**Comment by mnordine** [comment]
I think this may be a Chromium bug. Memory seems to be getting reclaimed in Chrome after dart2js.

**Comment by mnordine** [comment]
Although I'm still getting memory leaks in my more complex apps when loading/unloading texture atlases, even in Chrome

**Comment by bp74** [comment]
Yes i did a few repro cases and i saw it too. I concentrated on the EventListener that are not garbage collected, if the repro case is super simple it works - but once the code gets a little bit more complicated they are not collected anymore. Next step will be a JavaScript test to see if the behavior is similar, or if it an issue with dart/dart2js or the dart:html package.
